plugin/setup.go

```diff
@@ -0,0 +1,37 @@
+package lighthouse
+
+import (
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/mholt/caddy"
+)
+
+// init registers this plugin within the Caddy plugin framework. It uses "example" as the
+// name, and couples it to the Action "setup".
+func init() {
+	caddy.RegisterPlugin("lighthouse", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setupLighthouse,
+	})
+}
+
+// setup is the function that gets called when the config parser see the token "lighthouse". Setup is responsible
+// for parsing any extra options the this plugin may have. The first token this function sees is "lighthouse".
+func setupLighthouse(c *caddy.Controller) error {
+	c.Next() // Ignore "lighthouse" and give us the next token.
+	if c.NextArg() {
+		// If there was another token, return an error, because we currently don't have any configuration.
+		// Any errors returned from this setup function should be wrapped with plugin.Error, so we
+		// can present a slightly nicer error message to the user.
+		return plugin.Error("lighthouse", c.ArgErr())
+	}
+
+	// Add the Plugin to CoreDNS, so Servers can use it in their plugin chain.
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		return Lighthouse{Next: next}
```

Comment:
I think you want to return `& Lighthouse(...)`

Reply:
Used this as reference:
https://github.com/coredns/example/blob/master/setup.go#L41

Reply:
Yes but we agreed to use ginkgo, which is just a different framework. I believe you can map the same thing to ginkgo.

Reply:
This reply looks out of place here. did you mean this for the setup_test.go? If yes, I'll be making the switch to ginkgo in subsequent patch, if okay with you.